### PR TITLE
ISPN-7528 Store CacheTransaction.keyRead(...) in RepeatableReadEntry

### PR DESCRIPTION
**Pull request by rvansa**
https://issues.jboss.org/browse/ISPN-7528

Depends on #4905

### CODE REVIEW

File: core/src/main/java/org/infinispan/container/entries/MVCCEntry.java [hunk 1]
```diff
@@ -49,4 +49,32 @@ default void copyForUpdate() {}
     * source.
     */
    void updatePreviousValue();
+
+   /**
+    * Update initial value to the current value. This is executed when the value is loaded from an external location
+    * (such as cache store) before the first invocation of the command. Afterwards, {@link #skipLookup()} should be
+    * sent to true and therefore this will not be invoked.
```
Comment (pruivo) [inline core/src/main/java/org/infinispan/container/entries/MVCCEntry.java]: typo: ```sent``` => ```set```

**Comment by rvansa** [comment]
@pruivo Done. Btw., I think that we'll eventually make the EntryFactory return the entries in the context; I've come upon several situations when this will be handy.

**Comment by pruivo** [comment]
@rvansa :+1:  about the EntryFactory returning the entries 
CI failures doesn't seem to be caused by this PR (it is not the first failure this month). Going to push it.

File: core/src/main/java/org/infinispan/transaction/xa/CacheTransaction.java [hunk 2]
```diff
@@ -114,8 +115,16 @@ default void putLookedUpRemoteVersion(Object key, EntryVersion version) {}
    @Deprecated
    default EntryVersion getLookedUpRemoteVersion(Object key) { return null; }
 
+   /**
+    * @deprecated  since 9.1 Use {@link MVCCEntry#isRead()} instead
```
Comment (danberindei) [inline core/src/main/java/org/infinispan/transaction/xa/CacheTransaction.java]: Wrong `@since` version :)

Comment (rvansa) [inline core/src/main/java/org/infinispan/transaction/xa/CacheTransaction.java]: Oops, I am not used to getting my PRs merged in less than couple of months :)

**Pull request merged**